### PR TITLE
feat(verified-fetch): improve UX

### PR DESCRIPTION
**Pull request by SgtPooki**
Slightly modifies the code from https://github.com/ipfs-examples/helia-examples/pull/285 to have better UX.

Also implements the previously unimplemented `onFetchImage` and `onFetchFile` functions.

Better error messaging and loading messages.

Adds `fetch-auto` functionality to attempt to automatically determine content-type for proper rendering.

https://github.com/ipfs-examples/helia-examples/assets/1173416/68ffe8c8-c66d-42d5-98b2-af6209f619a7

### CODE REVIEW

File: examples/helia-browser-react-verified-fetch/src/App.tsx [hunk 1]
```diff
@@ -1,25 +1,182 @@
-import { useCallback, useState } from 'react'
+// import { verifiedFetch, createVerifiedFetch } from '@helia/verified-fetch'
```
Comment (SgtPooki) [inline examples/helia-browser-react-verified-fetch/src/App.tsx:1]: when https://github.com/ipfs/helia/pull/423 is merged, we should switch the auto content type button to use the `contentTypeParser` option.

File: examples/helia-browser-react-verified-fetch/src/App.tsx [hunk 2]
```diff
@@ -1,25 +1,182 @@
...
+  }, [])
+  const setError = useCallback((message: string) => {
+    setOutput('')
+    setLoadingTo(null)
+    setErr(message)
+  }, [])
+  const setLoading = useCallback((message: string) => {
+    setErr('')
+    setLoadingTo(loadingIndicator(message))
+  }, [])
```
Comment (SgtPooki) [inline examples/helia-browser-react-verified-fetch/src/App.tsx:60]: abstraction on top of the three potential displays that can happen. If you set loading, it should show that. If you set an error, it should show that. If you set output, it should show that.

Using these ensure the implicit rendering of content is handled appropriately, but we could use a `success`, `error`, and `loading` state and pass all three to `renderOutput`

**Review by SgtPooki — COMMENTED** [review]
self review

File: examples/helia-browser-react-verified-fetch/src/App.tsx [hunk 3]
```diff
@@ -1,25 +1,182 @@
...
+    } catch (err) {
+      setError((err as Error).message)
+
+    }
+  }, [])
+
+  const handleVideoType = useCallback(async (resp: Response) => {
+    try {
+      setLoading('Waiting for full video data...')
+      const blob = await resp.blob()
```
Comment (2color) [inline examples/helia-browser-react-verified-fetch/src/App.tsx:87]: At some point, I'd love to dig into streaming the video, but I think we might have to do that in a service worker.

**Review by 2color — APPROVED** [review]
Thanks for all of that. Not sure why I still can't get this to run locally (same error with verified-fetch looking for process). But all changes look reasonable to me. Merging.

**Pull request merged**